cmd/silkworm.cpp

```diff
@@ -14,291 +14,109 @@
-    prune_opts
-        .add_option("--prune", prune_mode,
-                    "Delete data older than 90K blocks (see \"--prune.*.older\" for different height)\n"
-                    "h - prune history (ChangeSets, HistoryIndices - used by historical state access)\n"
-                    "r - prune receipts (Receipts, Logs, LogTopicIndex, LogAddressIndex - used by eth_getLogs and "
-                    "similar RPC methods)\n"
-                    "t - prune transaction by it's hash index\n"
-                    "c - prune call traces (used by trace_* methods)\n"
-                    "If item is NOT in the list - means NO pruning for this data.\n"
-                    "Example: --prune=hrtc (default: none)",
-                    true)
-        ->check(PruneModeValidator());
+#include "common.hpp"
 
-    prune_opts.add_option("--prune.h.older", "Override default 90k blocks of history to prune")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.r.older", "Override default 90k blocks of receipts to prune")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.t.older", "Override default 90k blocks of transactions to prune")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.c.older", "Override default 90k blocks of call traces to prune")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.h.before", "Prune history data before this block")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.r.before", "Prune receipts data before this block")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.t.before", "Prune transactions data before this block")
-        ->check(CLI::Range(0u, UINT32_MAX));
-    prune_opts.add_option("--prune.c.before", "Prune call traces data before this block")
-        ->check(CLI::Range(0u, UINT32_MAX));
-
-    // Logging options
-    auto& log_opts = *cli.add_option_group("Log", "Logging options");
-    log_opts.add_option("--log.verbosity", log_settings.log_verbosity, "Sets log verbosity", true)
-        ->check(CLI::Range(static_cast<uint32_t>(log::Level::kCritical), static_cast<uint32_t>(log::Level::kTrace)))
-        ->default_val(std::to_string(static_cast<uint32_t>(log_settings.log_verbosity)));
-    log_opts.add_flag("--log.stdout", log_settings.log_std_out, "Outputs to std::out instead of std::err");
-    log_opts.add_flag("--log.nocolor", log_settings.log_nocolor, "Disable colors on log lines");
-    log_opts.add_flag("--log.utc", log_settings.log_utc, "Prints log timings in UTC");
-    log_opts.add_flag("--log.threads", log_settings.log_threads, "Prints thread ids");
-    log_opts.add_option("--log.file", log_settings.log_file, "Tee all log lines to given file name");
+using namespace silkworm;
 
-    cli.parse(argc, argv);
+static std::atomic<size_t> s_allocated_memory{0};
```

Comment:
What about memory allocated not with `operator new`, but with `malloc` itself? And I don't like operator overloading, feels too hacky. Using something like `malloc_stats` would probably be cleaner. I'd suggest to scrap allocated memory counting for now and do it later in a separate PR.

Reply:
This is a quick and dirty way to track the heap in a **portable** way. Is not meant to be definitive and is useful to me to track the sync loop I'm dealing with. Will gladly scrap it all when ready for review.
Besides **our** code does not make use of `malloc` (submodules do - but this is outside the scope of my actual nvestigations)

Reply:
`malloc_stats` is also *nix only

Reply:
Removed by 1fef176